helpers/external_acl/eDirectory_userip/ext_edirectory_userip_acl.cc

```diff
@@ -721,7 +721,11 @@ BindLDAP(edui_ldap_t *l, char *dn, char *pw, unsigned int t)
             /* We got a basedn, but it's not part of dn */
             xstrncpy(l->dn, dn, sizeof(l->dn));
             strncat(l->dn, ",", FREE_SPACE(l->dn));
-            strncat(l->dn, l->basedn, FREE_SPACE(l->dn));
+            const auto dnFreeBytes = FREE_SPACE(l->dn);
+            const auto dnAppendBytes = dnFreeBytes < strlen(l->basedn) ? dnFreeBytes : strlen(l->basedn);
+            const auto dnOldLength = strlen(l->dn);
+            memmove(l->dn + dnOldLength, l->basedn, dnAppendBytes);
```

Comment:
I had to do this change resolving GCC errors with '-O2'. The compiler thinks that buffers overlap:

```
error: ‘char* __builtin___strncat_chk(char*, const char*, long unsigned int, long unsigned int)’ accessing between 16393 and 8171 bytes at offsets 16416 and 24 may overlap 1 byte at offset 16416 [-Werror=restrict]
```

However, in reality, they do not overlap. This code compiles without optimization. I made a standalone test with strncat(), the error is produced when string arrays are stored within a structure:

```
struct Bar
{
  char src[8] = "aaaaaaa";
  char dst[8] = "bbbbbbb";
};
int main()
{   
  Bar bar;
  strncat(bar.dst, bar.src, sizeof(bar.dst - strlen(bar.dst) - 1);
  return 0;
}
```

Note that there is no such error when the arrays are declared within main().